apps/shared/avifpng.c

```diff
@@ -17,6 +18,106 @@ typedef png_bytep png_iccp_datap;
 typedef png_charp png_iccp_datap;
 #endif
 
+// Converts a hexadecimal string which contains 2-byte character representations of hexadecimal values to raw data.
+// hexString may contain values consisting of [A-F][a-f][0-9] in pairs, e.g., 7af2..., separated by any number of newlines.
+// On success the bytes are filled and AVIF_TRUE is returned.
+// AVIF_FALSE is returned if fewer than expectedLength hexadecimal pairs are converted.
+static avifBool avifHexStringToBytes(const char * hexString, size_t expectedLength, avifRWData * bytes)
+{
+    avifRWDataRealloc(bytes, expectedLength);
+    const char * src = hexString;
+    uint8_t * dst = bytes->data;
+    size_t actualLength = 0;
+    for (; (actualLength < expectedLength) && (*src != '\0'); ++actualLength, ++dst) {
+        while (*src == '\n') {
+            ++src;
+        }
+        const char hexValue[3] = { src[0], src[1], '\0' };
+        src += 2;
+        char * end;
+        bytes->data[actualLength] = (uint8_t)strtol(hexValue, &end, 16);
```

Comment:
If `hexValue` is `{ '+', '1', '\0' }` or `{ ' ', '1', '\0' }`, the code will convert it to `1` successfully. But this kind of input is malformed, right?